src/main/java/fi/dy/masa/servux/dataproviders/StructureDataProvider.java

```diff
@@ -372,7 +450,8 @@ protected boolean chunkHasStructureReferences(int chunkX, int chunkZ, World worl
 
     protected void sendStructures(ServerPlayerEntity player,
                                   Map<Structure, LongSet> references,
-                                  int tickCounter)
+                                  int tickCounter,
+                                  boolean useApi)
```

Comment:
Perhaps, the `useApi` could be renamed to better resemble the fact that it's counting failure attempts and then cancels furher send attempts?

Maybe something along these `ignoreFailure` or `useCountingEncoder`

This could also be applied to the [`ServuxStructuresHandler#encodeNbtCompound()`](https://github.com/sakura-ryoko/servux/blob/b2a63a58292bcfab3548af30e0c465defb1a4ec8/src/main/java/fi/dy/masa/servux/network/handler/ServuxStructuresHandler.java#L113) method